### ISSUE TITLE
error C2872: “ACCESS_MASK”:

### ISSUE THREAD

**Issue by xdg2016** [oylz/DS#13]
Hello,thanks for your code. when I build the project in vs2013, there are some errors:
![image](https://user-images.githubusercontent.com/16351823/55222742-9657c180-5247-11e9-978e-32dc8edcd12d.png)
I searched online for a long time, but the solutions didn't work. Can you help me ?

**Comment by oylz** [comment]
sorry,I do not have windows OS right now,I have some guess:
0.the order of included header files
1.the vs2013 has not been installed completely 
2.may be the c++11 cause,you should install vs2015 or later

**Comment by xdg2016** [comment]
Thank you,I have addressed the proplem, but I meet a new error
![image](https://user-images.githubusercontent.com/16351823/55694236-d3f5e080-59e5-11e9-85f4-1949ec8182a6.png)
I can't find any solution to this proplem,and I can't go next.

**Comment by oylz** [comment]
there is different  template class implement  between vc and g++.
in vc, we should implement all in  template class header file.
just like:
* move all implement from [matrix cpp](https://github.com/oylz/DS/blob/master/deepsort/munkres/matrix.cpp) to [matrix header](https://github.com/oylz/DS/blob/master/deepsort/munkres/matrix.h)

* do the same, if you hit the same problem in other classes

**Comment by xdg2016** [comment]
Thanks a lot! I can run it. I add the YOLOv3 dll to track a vedio, and I find a small bug in your code:
![image](https://user-images.githubusercontent.com/16351823/55848848-fe2fd580-5b80-11e9-8ae2-949e01ad91ec.png)
you didn't check the bound of the ROI, this can result in the Mat out of range. I modify the code to address the problem. Now the code runs successfully. 
Thanks again!

**Comment by BigTree-git** [comment]
> Hello,thanks for your code. when I build the project in vs2013, there are some errors:
> ![image](https://user-images.githubusercontent.com/16351823/55222742-9657c180-5247-11e9-978e-32dc8edcd12d.png)
> I searched online for a long time, but the solutions didn't work. Can you help me ?



> Hello,thanks for your code. when I build the project in vs2013, there are some errors:
> ![image](https://user-images.githubusercontent.com/16351823/55222742-9657c180-5247-11e9-978e-32dc8edcd12d.png)
> I searched online for a long time, but the solutions didn't work. Can you help me ?

**Comment by BigTree-git** [comment]
@xdg2016   hello  i  have some same problem with you ,how you slove  error C2872: “ACCESS_MASK”?

**Comment by Vincent630** [comment]
> Hello,thanks for your code. when I build the project in vs2013, there are some errors:
> ![image](https://user-images.githubusercontent.com/16351823/55222742-9657c180-5247-11e9-978e-32dc8edcd12d.png)
> I searched online for a long time, but the solutions didn't work. Can you help me ?
i want to build  in vs2019,can you tell me some details about this?

**Comment by BigTree-git** [comment]
sorry  i still face same problem   i quit.....




------------------ 原始邮件 ------------------
发件人: "Vincent630"<notifications@github.com>;
发送时间: 2019年7月25日(星期四) 中午11:08
收件人: "oylz/DS"<DS@noreply.github.com>;
抄送: "蓝色日出"<604976301@qq.com>;"Comment"<comment@noreply.github.com>;
主题: Re: [oylz/DS] error C2872: “ACCESS_MASK”: (#13)



 
Hello,thanks for your code. when I build the project in vs2013, there are some errors:
 
 I searched online for a long time, but the solutions didn't work. Can you help me ?
 i want to build  in vs2019,can you tell me some details about this?

**Comment by Vincent630** [comment]
> sorry i still face same problem i quit.....
> […](#)
> ------------------ 原始邮件 ------------------ 发件人: "Vincent630"<notifications@github.com>; 发送时间: 2019年7月25日(星期四) 中午11:08 收件人: "oylz/DS"<DS@noreply.github.com>; 抄送: "蓝色日出"<604976301@qq.com>;"Comment"<comment@noreply.github.com>; 主题: Re: [oylz/DS] error C2872: “ACCESS_MASK”: (#13) Hello,thanks for your code. when I build the project in vs2013, there are some errors: I searched online for a long time, but the solutions didn't work. Can you help me ? i want to build in vs2019,can you tell me some details about this? — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or mute the thread.

have you tmake it work on ubuntu platform?

**Comment by BigTree-git** [comment]
no,  but i find a strange problem,i'm not sure  whether it's useful to you ,you may try to check  winnt.h  in path "C:\Program Files (x86)\原始windowkit\8.1\Include\um".in my windows7  os,it show as:




if  you are same with me, maybe you shoule replace  windowkit


------------------ 原始邮件 ------------------
发件人: "Vincent630"<notifications@github.com>;
发送时间: 2019年7月25日(星期四) 下午2:49
收件人: "oylz/DS"<DS@noreply.github.com>;
抄送: "蓝色日出"<604976301@qq.com>;"Comment"<comment@noreply.github.com>;
主题: Re: [oylz/DS] error C2872: “ACCESS_MASK”: (#13)



 
sorry i still face same problem i quit.....
 …
 ------------------ 原始邮件 ------------------ 发件人: "Vincent630"notifications@github.com; 发送时间: 2019年7月25日(星期四) 中午11:08 收件人: "oylz/DS"DS@noreply.github.com; 抄送: "蓝色日出"604976301@qq.com;"Comment"comment@noreply.github.com; 主题: Re: [oylz/DS] error C2872: “ACCESS_MASK”: (#13) Hello,thanks for your code. when I build the project in vs2013, there are some errors: I searched online for a long time, but the solutions didn't work. Can you help me ? i want to build in vs2019,can you tell me some details about this? — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or mute the thread.
  
have you tmake it work on ubuntu platform?
